### PR TITLE
fix: use header height for Tendermint consensus state storage

### LINKED ISSUE #1080: [ICS07] Incorrect `ConsensusState` storage for headers with heights lower than `ClientState` height
## Bug Summary

Reported by @vmarkushin 

When a header with a height lower than the current client height is submitted for the first time, we store the consensus state [at the height of the current client state](https://github.com/cosmos/ibc-rs/blob/0a275ff15b4392f33cf0e16ad880f67ddbcdeca4/ibc-clients/ics07-tendermint/src/client_state.rs#L386-L387) instead of the provided header height. This action overwrites the existing consensus state for the current height. Since the header is from the past, a `ConsensusState` with an older root is installed at that path. This disrupts commitment proof verification, as the provided proof will be for a different state root than the one we actually stored.

## Details

- [H] Installed headers
- (H) Unseen headers

---- [5] ---- [7] -- (8) ---- [10] -------------- (20)

Scenario: Submit a header with height 8 and the trusted height of 7

## Version

=< v0.50.0

### PR DESCRIPTION
**Pull request by Farhad-Shabani**
Closes: #1080

### Along with this PR

- Removed the check and test for stale mock headers as it was not sufficiently general and did not provide meaningful validation.
- Nit: placing unclog of issue 835 under the `breaking-changes`

______

### PR author checklist:
- [X] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests.
- [X] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: ibc-testkit/tests/core/ics02_client/update_client.rs [hunk 1]
```diff
@@ -109,6 +109,81 @@ fn test_update_client_ok(fixture: Fixture) {
...
+
+    let mut ctx = MockContext::default().with_client_config(
+        MockClientConfig::builder()
+            .client_chain_id(chain_id_b.clone())
+            .client_type(tm_client_type())
+            .client_id(client_id.clone())
+            .latest_height(latest_height)
+            .build(),
+    );
+    let mut router = MockRouter::new_with_transfer();
```
Comment (seanchen1991) [inline ibc-testkit/tests/core/ics02_client/update_client.rs:131]: Small nit, but you can construct the `router` from the `fixture` function.

**Review by seanchen1991 — APPROVED** [review]
One small nit, otherwise looks good to me

File: ibc-testkit/tests/core/ics02_client/update_client.rs (hunk 1 above)
Comment (Farhad-Shabani) [reply ibc-testkit/tests/core/ics02_client/update_client.rs:131]: The thing is: I need a `MockContext` with the Tendermint `ClientState` for this test, but the current fixture isn't configurable. Getting the router from fixture means constructing an unnecessary `MockContext`.

As an aside, this fixture should be overhauled in favor of a better approach like the one we have for connection tests.

**Review by ancazamfir — APPROVED** [review]
lgtm!

File: ibc-testkit/tests/core/ics02_client/update_client.rs (hunk 1 above)
Comment (seanchen1991) [reply ibc-testkit/tests/core/ics02_client/update_client.rs:131]: > As an aside, this fixture should be overhauled in favor of a better approach like the one we have for connection tests.

I'm trying to address this by defining a `mock_context` function like so:
```
fn mock_context(
    client_id: ClientId,
    height: Height,
    client_chain_id: Option<ChainId>,
    client_type: Option<ClientType>,
) -> MockContext {
    let client_config_builder = MockClientConfig::builder();

    if let Some(client_chain_id) = client_chain_id {
        client_config_builder.client_chain_id(client_chain_id);
    }

    if let Some(client_type) = client_type {
        client_config_builder.client_type(client_type);
    }

    let client_config = client_config_builder
        .client_id(client_id)
        .latest_height(height)
        .build();

   MockContext::default().with_client_config(client_config)
}
```

However, the compiler complains and prescribes calling `clone` on every line that modifies the `MockClientConfigBuilder`, which is obviously not ideal. Is there a better way to go about defining a configurable `mock_context` function?

I'm not sure where the connection tests are located.

Comment (seanchen1991) [reply ibc-testkit/tests/core/ics02_client/update_client.rs:131]: Looks like fully refactoring the client update tests to follow the approach that the connection tests use is a much bigger undertaking that deserves its own issues and PR. I will open a separate issue for that.

Comment (Farhad-Shabani) [reply ibc-testkit/tests/core/ics02_client/update_client.rs:131]: Yes. Let's take care of this as a separate issue. Thank you Sean 🙏🏻

**Pull request merged**